mod/ArchConsoleManager.cs

```diff
@@ -0,0 +1,191 @@
+                ShowConsoles(isPaused);
+            }
+        }
+
+        // Creates the two console displays
+        private void CreateConsoles(OWScene scene, OWScene loadScene)
+        {
+            if (loadScene != OWScene.SolarSystem && loadScene != OWScene.EyeOfTheUniverse) return;
+            // Create objects and establish references
+            gameplayConsoleEntries = new List<string>();
+            console = GameObject.Instantiate(Randomizer.Assets.LoadAsset<GameObject>("ArchRandoCanvas"));
+            pauseConsoleVisuals = console.transform.Find("PauseConsole").gameObject;
+            pauseConsole = console.transform.Find("PauseConsole/Scroll View/Viewport/PauseConsoleText").gameObject;
+            gameplayConsole = console.transform.Find("GameplayConsole/GameplayConsoleText").gameObject;
+            pauseConsoleText = pauseConsole.GetComponent<Text>();
+            gameplayConsoleText = gameplayConsole.GetComponent<Text>();
+
+            pauseConsoleText.text = string.Empty;
+            gameplayConsoleText.text = string.Empty;
+
+            // Copy text over from previous loops
+            foreach (string entry in consoleHistory)
+            {
+                AddText(entry, true, true);
+            }
+            console.GetComponentInChildren<InputField>().onEndEdit.AddListener(OnConsoleEntry);
+            consoleText = console.GetComponentInChildren<InputField>();
+            pauseConsoleVisuals.SetActive(false);
+
+            StartCoroutine(LoopGreeting());
+        }
+
+        // Shows the appropriate consoles when the game is paused or not
+        private void ShowConsoles(bool showPauseConsole)
+        {
+            pauseConsoleVisuals.SetActive(showPauseConsole);
+            gameplayConsole.SetActive(!showPauseConsole);
+        }
+
+        /// <summary>
+        /// Adds a new text entry to the in-game consoles
+        /// </summary>
+        /// <param name="text">The text to add to the consoles</param>
+        /// <param name="skipGameplayConsole">Whether to only show text on the pause console</param>
+        /// <param name="skipHistory">Whether to not save this text between loops</param>
+        public void AddText(string text, bool skipGameplayConsole = false, bool skipHistory = false)
+        {
+            if (!skipHistory) consoleHistory.Add(text);
+            pauseConsoleText.text += "\n" + text;
+
+            // We don't need to bother editing the Gameplay Console if this is on
+            if (!skipGameplayConsole)
+            {
+                if (gameplayConsoleEntries.Count < 6)
+                {
+                    gameplayConsoleEntries.Add("\n" + text);
+                }
+                else
+                {
+                    for (int i = 0; i < 5; i++)
+                    {
+                        gameplayConsoleEntries[i] = gameplayConsoleEntries[i + 1];
+                    }
+                    gameplayConsoleEntries[5]= "\n" + text;
```

Comment:
Hm, would it be simpler to use a `Queue<string>` instead of a `List<string>`? Not sure if a queue would interfere with other parts of this.

Reply:
I have never used a Queue collection before, I'll have to look at how that works.

Reply:
Queue implemented